go-controller/hybrid-overlay/pkg/controller/node_linux.go

```diff
@@ -146,12 +146,24 @@ func (n *NodeController) addOrUpdatePod(pod *kapi.Pod, ignoreLearn bool) error {
 				"actions=set_field:%s->eth_src,set_field:%s->eth_dst,output:ext",
 			cookie, podIP.IP, n.drMAC.String(), podMAC))
 
-		if namespaceExternalGw == "" || namespaceVTEP == "" {
+		if namespaceExternalGwRaw == "" || namespaceVTEPRaw == "" {
 			klog.Infof("Hybrid Overlay Gateway mode not enabled for pod %s, namespace does not have hybrid"+
-				"annotations, external gw: %s, VTEP: %s", pod.Name, namespaceExternalGw, namespaceVTEP)
+				"annotations, external gw: %s, VTEP: %s", pod.Name, namespaceVTEPRaw, namespaceVTEPRaw)
 			n.updateFlowCacheEntry(cookie, flows, ignoreLearn)
 			continue
 		}
+		// validate the external gateway is a valid IP address
+		namespaceExternalGwIP := net.ParseIP(namespaceExternalGwRaw)
+		if namespaceExternalGwIP == nil {
+			return fmt.Errorf("failed to validate a valid external gateway ip address %s: %v", namespaceExternalGwRaw, err)
+		}
+		namespaceExternalGw := namespaceExternalGwIP.String()
+		// validate the vtep is a valid IP address
+		namespaceVTEPIP := net.ParseIP(namespaceVTEPRaw)
+		if namespaceVTEPIP == nil {
+			return fmt.Errorf("failed to validate a valid vtep ip address %s: %v", namespaceVTEPRaw, err)
+		}
+		namespaceVTEP := namespaceVTEPIP.String()
 
 		portMACRaw := strings.Replace(n.drMAC.String(), ":", "", -1)
 		vtepIPRaw := getIPAsHexString(net.ParseIP(namespaceVTEP))
```

Comment:
Note that `namespaceVTEP` gets reparsed twice later in this function so you should fix that to use the already-parsed version